tracer/test/Datadog.Trace.TestHelpers/TracerHelper.cs

```diff
@@ -15,14 +17,36 @@ internal static class TracerHelper
         /// <summary>
         /// Create a test instance of the Tracer, that doesn't used any shared instances
         /// </summary>
-        public static Tracer Create(
+        public static ScopedTracer Create(
             TracerSettings settings = null,
             IAgentWriter agentWriter = null,
             ITraceSampler sampler = null,
             IScopeManager scopeManager = null,
             IDogStatsd statsd = null)
         {
-            return new Tracer(settings, agentWriter, sampler, scopeManager, statsd);
+            return new ScopedTracer(settings, agentWriter, sampler, scopeManager, statsd);
+        }
+
+        /// <summary>
+        /// Create a test instance of the Tracer, that doesn't used any shared instances
+        /// </summary>
+        public static ScopedTracer CreateWithFakeAgent(
+            TracerSettings settings = null)
+        {
+            return new ScopedTracer(settings, Mock.Of<IAgentWriter>());
+        }
+
+        public class ScopedTracer : Tracer, IDisposable
+        {
+            public ScopedTracer(TracerSettings settings = null, IAgentWriter agentWriter = null, ITraceSampler sampler = null, IScopeManager scopeManager = null, IDogStatsd statsd = null)
+                : base(settings, agentWriter, sampler, scopeManager, statsd)
+            {
+            }
+
+            public void Dispose()
+            {
+                TracerManager.ShutdownAsync().GetAwaiter().GetResult();
```

Comment:
Isn't `TraceManager` shared? What happens when there are multiple `ScopedTracer` instances and once of them is disposed?

Reply:
That's the local TracerManager, not the static one. It's not supposed to be shared (and when we do, we disable concurrency on those tests)

Reply:
(that's the one reason why I think prefixing all members with `this` is better than prefixing only fields with `_`. The latter causes ambiguity for properties)

Reply:
Oops, my mistake. Thanks for clarifying.